### DISCUSSION TITLE
v10 is breaking for us so we're stuck on v9 (follow up on release notes 10.0.0)

### DISCUSSION THREAD

**Discussion by tyler-g** [bbc/sqs-consumer#491] [Help] [answered]
### Summary

Hey all, we saw this was already mentioned for the breaking change in v10:

[v10.0.0](https://github.com/bbc/sqs-consumer/releases/tag/v10.0.0)
Potentially Breaking Changes
SQS Consumer has been transitioned to use ESM in its codebase, in order to support this, we have had to change how we deploy the package in order to build both ESM and CJS versions of the package.

This may cause issues with your software, if it does, please let us know.

This has forced us to be stuck on v9 because on v10 it will not find the type index file in sqs-consumer package.  Wanted to check if there is any follow up to this that will eventually let us upgrade, or if we'll be stuck on v9.   Thanks

**Comment by nicholasgriffintn** [comment]
Hmm, maybe my idea of how types worked was wrong.

I'll add it to my list to have a look into this, unless anyone knows the fix and fancies putting a pr in before I get the chance to.

**Reply by tyler-g** [reply]
Thanks!  Not really a dealbreaker right now but just looking ahead – I'd imagine others are in a similar situation

**Reply by nicholasgriffintn** [reply]
Thanks, at least one possibility is that this is happening for people using `moduleResolution`: `node` still.

I've published a canary version that adds `main` and `types` back, it's hoped that this might resolve the issue, I've not fully tested yet though: https://github.com/bbc/sqs-consumer/issues/492#issuecomment-2083611566

**Reply by nicholasgriffintn** [reply, marked as answer]
Update a fix for this was merged in 10.2.1.
